From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 2 Jan 2012 02:50:42 +0000
Subject: [PATCH] ARM: Remove use of possibly-undefined BUILD_BUG_ON in
 <asm/bug.h>

<asm/bug.h> may be included by either <linux/bug.h> or
<linux/kernel.h> but only the latter will define BUILD_BUG_ON.
Bodge it until there is a proper upstream fix.
---
 arch/arm/include/asm/bug.h |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/arch/arm/include/asm/bug.h b/arch/arm/include/asm/bug.h
index 9abe7a0..fac79dc 100644
--- a/arch/arm/include/asm/bug.h
+++ b/arch/arm/include/asm/bug.h
@@ -32,7 +32,6 @@
 
 #define __BUG(__file, __line, __value)				\
 do {								\
-	BUILD_BUG_ON(sizeof(struct bug_entry) != 12);		\
 	asm volatile("1:\t" BUG_INSTR_TYPE #__value "\n"	\
 		".pushsection .rodata.str, \"aMS\", %progbits, 1\n" \
 		"2:\t.asciz " #__file "\n" 			\
-- 
1.7.7.3

